File: Programmers/Level4/chooseCity/ChooseCity.java
import java.util.Arrays;
import java.util.Comparator;

class ChooseCity
{
    public int chooseCity(int n, int [][]city)
    {
        int answer = 0;
        int sum=0;
        for(int i=0; i<city.length; ++i)
        {
            sum += city[i][1];
        }
        int half = sum/2;
        if(sum%2 == 1)
        {
            ++half;
        }


        Arrays.sort(city,new Comparator<int[]>(){
            @Override
            public int compare(final int[] arr1, final int[] arr2)
            {
                if(arr1[1] - arr2[1] > 0)
                {
                    return -1;
                }
                else return 1;
            }
        });


        int temp=0;
        for(int i=0; i<city.length; ++i)
        {
            temp += city[i][1];
            if(temp >= half)
            {
                answer = city[i][0];
                break;
            }
        }
        return answer;
    }
    public static void main(String[] args)
    {
        ChooseCity test = new ChooseCity();
        int tn = 3;
        int [][]tcity = {{1,5},{2,2},{3,3}};
        System.out.println(test.chooseCity(tn,tcity));
    }
}

File: Programmers/Level4/chooseCity/chooseCity.py
def chooseCity(n,city):
    answer = 0
    city.sort()
    sum_val = sum([i[1] for i in city])
    if sum_val % 2 == 0:
        half = sum_val // 2
    else:
        half = sum_val // 2 +1
    temp = 0
    for i in range(n):
        temp += city[i][1]
        if temp>=half:
            answer=city[i][0]
            break;
    return answer

#아래 코드는 출력을 위한 테스트 코드입니다.

print(chooseCity(3,[[1,5],[2,2],[3,3]]))